salt/modules/zypper.py

```diff
@@ -56,6 +56,21 @@ def __virtual__():
     return __virtualname__
 
 
+def _zypper(as_list=False):
+    '''
+    Return zypper command with default options as a string.
+
+    CMD: zypper --non-interactive
+
+    as_list:
+        if set to True, the command and the default options
+        are returned as a list
+    '''
+    if as_list:
+        return ['zypper', '--non-interactive']
+    return "zypper --non-interactive "
```

Comment:
Further look shows that `_zypper()` is now concatenating strings. :cry: This is not good and doesn't take the advantage of having a function, since some commands are _concatenated_ and others are _formatted_. This is kind of messy. To unify it, if something needs to be added to the `_zypper()` output, I'd rather keep it that way:

```python
def _zypper(*opts, **kw):
    '''
    Doc here
    '''
    cmd = ['zypper', '--non-interactive']
    cmd.extend(opts)
    
    return kw.get('as_list', False) and cmd or ' '.join(cmd)
```

The usage would be the following:
```python
#  Before
cmd = '{0} info -t package {1}'.format(_zypper(), ' '.join(batch[:batch_size]))

# After:
cmd = _zypper('info', '-t', 'package', ' '.join(batch[:batch_size])
```

Or avoid direct concatenation:
```python
# Before
cmd = _zypper() + 'update --auto-agree-with-licenses'

# After
cmd = _zypper('update', '--auto-agree-with-licenses')
```

In case to get it as list, pass the same variable `as_list=True`:
```python
cmd = _zypper('update', '--auto-agree-with-licenses', as_list=True)
```